tsdb/store.go

```diff
@@ -1709,7 +1709,9 @@ func (s *Store) TagValues(ctx context.Context, auth query.Authorizer, shardIDs [
 			switch e.Op {
 			case influxql.EQ, influxql.NEQ, influxql.EQREGEX, influxql.NEQREGEX:
 				tag, ok := e.LHS.(*influxql.VarRef)
-				if !ok || tag.Val != "_name" {
+				// Drop all system names except "_name" so that regex and negation
+				// operators work.
+				if !ok || (tag.Val != "_name" && influxql.IsSystemName(tag.Val)) {
```

Comment:
Rather than swallowing everything except `_name` fields, this will swallow just system names that aren't `_name`. This allows the call to `MeasurementNamesByExpr` below to work properly, since it can only work with `_name` and no other system names.